scipy/spatial/transform/rotation.py

```diff
@@ -131,54 +132,58 @@ def as_dcm(self):
             return dcm
 
     @classmethod
-    def from_dcm(cls, mat, scalar=True):
-        # For now, mat must be a 3D matrix
-        num_rotations = mat.shape[0]
-        if scalar:
-            quat = np.empty((num_rotations, 4))
-            for rot_num in range(num_rotations):
-                A = mat[rot_num]
-                trA = A.trace()
-                choice = np.argmax(np.array([A[0, 0], A[1, 1], A[2, 2], trA]))
-
-                if choice in [0, 1, 2]:
-                    i = choice
-                    j = (i + 1) % 3
-                    k = (j + 1) % 3
-                    quat[rot_num, i] = 1 - trA + 2 * A[i, i]
-                    quat[rot_num, j] = A[j, i] + A[i, j]
-                    quat[rot_num, k] = A[k, i] + A[i, k]
-                    quat[rot_num, 3] = A[k, j] - A[j, k]
-                else:
-                    quat[rot_num] = np.array([
-                        A[2, 1] - A[1, 2],
-                        A[0, 2] - A[2, 0],
-                        A[1, 0] - A[0, 1],
-                        1 + trA])
+    def from_dcm(cls, dcm):
+        """Initialize rotation from direction cosine matrix.
+
+        This classmethod return a `Rotation` object from the input direction
+        cosine matrices. If the input matrix is not orthogonal, this method
+        creates the best approximation of the rotation.
+
+        Parameters
+        ----------
+        dcm : array_like, shape (N, 3, 3) or (3, 3)
+            Each `dcm[i]` corresponds to a `(3, 3)` direction cosine matrix.
+        """
+        is_single = False
+        dcm = np.asarray(dcm, dtype=float)
+
+        if dcm.ndim not in [2, 3] or (dcm.shape[-2], dcm.shape[-1]) != (3, 3):
+            raise ValueError("Expected `dcm` to have shape (3, 3) or "
+                             "(N, 3, 3), got {}".format(dcm.shape))
+
+        # If a single dcm is given, convert it to 3D 1 x 3 x 3 matrix but set
+        # self._single to True so that we can return appropriate objects in
+        # the `to_...` methods
+        if dcm.shape == (3, 3):
+            dcm = dcm.reshape((1, 3, 3))
+            is_single = True
+
+        num_rotations = dcm.shape[0]
+
+        decision_matrix = np.zeros((num_rotations, 4))
```

Comment:
This is totally minor, but technically you can use `empty` here. I think it makes sense logically as you fill each element explicitly afterwards.

Reply:
I'm filling out `decision_matrix[:, :3]` explicitly, and the the last column containing the trace is calculated by summing each row `decision_matrix[:, -1] = decision_matrix.sum(axis=1)`. For this to work, we need the last column to have zeros initially.

Reply:
Right, it's sort of tricky then. Perhaps rewrite as `decision_matrix[:, :3].sum(axis=1)`?

Reply:
Sure.